### ISSUE TITLE
Wrong rendering of DayToSecond literals with negative components

### ISSUE THREAD

**Issue by knutwannheden** [jOOQ/jOOQ#9202]
(This issue was originally reported as part of #9201.)

I'm not totally sure if the DayToSecond class really intends for components to be negative, but it permits negative values passed to its public constructors.

    new DayToSecond(0, -1)

And rendered to SQL for PostgreSQL, this produces:
    
    cast('+0 0-1:00:00.000000000' as interval day to second)
    
Note the leading `0` character before the hours, which is included [because -1 < 10](https://github.com/jOOQ/jOOQ/blob/28677b252e56b7547fda125155deebdc097cf98a/jOOQ/src/main/java/org/jooq/types/DayToSecond.java#L552).  And of course, PostgreSQL rejects this SQL: `invalid input syntax for type interval: "+0 0-1:00:00.000000000"`
    
Similar issues occur when the other components are negative, e.g.

    new DayToSecond(-1, -1, -1, -1)

    cast('+-1 0-1:0-1:0-1.000000000' as interval day to second)

These issues may occur in dialects other than PostgreSQL - I've only tested with PostgreSQL.

**Comment by lukaseder** [comment]
I agree with this change, although it might be perceived as a backwards incompatible change if someone relies on the current behaviour, so we will not backport it to versions prior to 3.13

**Comment by knutwannheden** [comment]
In this context the logic in `PostgresUtil#toYearToMonth()` and `#toDayToSecond()` must also be looked at, as it contains this problematic statement, which checks if any of the components is negative:

```java
        boolean negative = pgInterval.toString().contains("-");
```
